pkg/cache/internal/options.go

```diff
@@ -0,0 +1,17 @@
+package internal
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Options are the optional arguments for creating a new InformersMap object
+type Options struct {
+	Scheme                  *runtime.Scheme
+	Mapper                  meta.RESTMapper
+	Resync                  *time.Duration
+	Namespace               string
+	FieldSelectorByResource map[string]string
+}
```

Comment:
Given that this is all internal, how are users supposed to use this new feature?

Reply:
Removed the internal Options